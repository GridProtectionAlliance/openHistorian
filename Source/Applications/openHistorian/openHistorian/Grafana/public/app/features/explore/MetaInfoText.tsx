import { css } from '@emotion/css';
import React, { memo } from 'react';

import { GrafanaTheme2 } from '@grafana/data';
import { useStyles2 } from '@grafana/ui';

const getStyles = (theme: GrafanaTheme2) => ({
  metaContainer: css`
    flex: 1;
    color: ${theme.colors.text.secondary};
    margin-bottom: ${theme.spacing(2)};
    min-width: 30%;
    display: flex;
    flex-wrap: wrap;
  `,
  metaItem: css`
    margin-right: ${theme.spacing(2)};
    margin-top: ${theme.spacing(0.5)};
    display: flex;
    align-items: baseline;

    .logs-meta-item__error {
      color: ${theme.colors.error.text};
    }
  `,
  metaLabel: css`
    margin-right: calc(${theme.spacing(2)} / 2);
    font-size: ${theme.typography.bodySmall.fontSize};
    font-weight: ${theme.typography.fontWeightMedium};
  `,
  metaValue: css`
    font-family: ${theme.typography.fontFamilyMonospace};
    font-size: ${theme.typography.bodySmall.fontSize};
  `,
});

export interface MetaItemProps {
  label?: string;
  value: string | JSX.Element;
}

const MetaInfoItem = memo(function MetaInfoItem(props: MetaItemProps) {
  const style = useStyles2(getStyles);
  const { label, value } = props;

  return (
    <div data-testid="meta-info-text-item" className={style.metaItem}>
      {label && <span className={style.metaLabel}>{label}:</span>}
      <span className={style.metaValue}>{value}</span>
    </div>
  );
});

interface MetaInfoTextProps {
  metaItems: MetaItemProps[];
}

export const MetaInfoText = memo(function MetaInfoText(props: MetaInfoTextProps) {
  const style = useStyles2(getStyles);
  const { metaItems } = props;

  return (
    <div className={style.metaContainer} data-testid="meta-info-text">
      {metaItems.map((item, index) => (
        <MetaInfoItem key={`${index}-${item.label}`} label={item.label} value={item.value} />
      ))}
    </div>
  );
});
