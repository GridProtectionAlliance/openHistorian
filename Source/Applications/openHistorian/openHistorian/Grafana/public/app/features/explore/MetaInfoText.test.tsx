import React from 'react';
import { shallow, render } from 'enzyme';
import { MetaInfoText, MetaItemProps } from './MetaInfoText';
describe('MetaInfoText', () => {
  it('should render component', () => {
    const items: MetaItemProps[] = [
      { label: 'label', value: 'value' },
      { label: 'label2', value: 'value2' },
    ];
    const wrapper = shallow(<MetaInfoText metaItems={items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render items', () => {
    const items: MetaItemProps[] = [
      { label: 'label', value: 'value' },
      { label: 'label2', value: 'value2' },
    ];

    const wrapper = render(<MetaInfoText metaItems={items} />);
    expect(wrapper.find('label')).toBeTruthy();
    expect(wrapper.find('value')).toBeTruthy();
    expect(wrapper.find('label2')).toBeTruthy();
    expect(wrapper.find('value2')).toBeTruthy();
  });

  it('should render no items when the array is empty', () => {
    const items: MetaItemProps[] = [];

    const wrapper = shallow(<MetaInfoText metaItems={items} />);
    expect(wrapper.find('div').exists()).toBeTruthy();
    expect(wrapper.find('div').children()).toHaveLength(0);
  });
});
