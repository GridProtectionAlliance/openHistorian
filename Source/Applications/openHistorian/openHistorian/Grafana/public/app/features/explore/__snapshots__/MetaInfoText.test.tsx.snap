// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaInfoText should render component 1`] = `
<div
  className="css-1jdkzq2"
>
  <Memo(MetaInfoItem)
    key="0-label"
    label="label"
    value="value"
  />
  <Memo(MetaInfoItem)
    key="1-label2"
    label="label2"
    value="value2"
  />
</div>
`;
